reconcile/utils/gql.py

```diff
@@ -108,18 +114,16 @@ def __init__(self, url, token=None, int_name=None, validate_schemas=False):
                 raise GqlApiIntegrationNotFound(int_name)
 
     @retry(exceptions=GqlApiError, max_attempts=5, hook=capture_and_forget)
-    def query(self, query, variables=None, skip_validation=False):
+    def query(self, query, variables=None, skip_validation=False) -> Dict[str, Dict]:
         try:
-            # supress print on HTTP error
-            # https://github.com/prisma-labs/python-graphql-client
-            # /blob/master/graphqlclient/client.py#L32-L33
-            with open(os.devnull, "w") as f, contextlib.redirect_stdout(f):
-                result_json = self.client.execute(query, variables)
+            result = self.client.execute(
+                gql(query), variables, get_execution_result=True
+            ).formatted
         except Exception as e:
+            # The client raises TransportQueryError if there is error returned with payload
+            # or AssertionError when no data is returned
             raise GqlApiError("Could not connect to GraphQL server ({})".format(e))
```

Comment:
I kept it as generic so as to avoid digging into implementation the next time we update the library if the contract changes for some reason. WDYT?
Yeah, the `AssertionError` was bit weird to me as well. I thought maybe this is a python thing :smile:

Reply:
I think that always returning "Could not connect" is misleading because some of these aren't connection issues. The advantage of returning our own exception is that when the implementation changes, like in this case, we don't leak that change to the clients. I'm not sure that we get that much more benefit from having a catch-all and a single exception message.

You probably don't need to catch `AssertionError`. In Python, `assert` is really only used for things that absolutely shouldn't be happening, or to do things like appease Mypy. You can read up more on why asserts aren't used much in Python and how certain optimizer flags can actually disable them. I would still suggest we catch the other exceptions specifically and keep the exception messages the same as they were before unless we have a good reason to change them. You can also catch `Exception` at the end, but catching the more specific exceptions first will replace the if statements that existed before.

Reply:
I decided to `AssertionError` as the underlying library is throwing it and it felt a bit odd not to catch it but others. I hope the library changes the implementation.

Let me know if this is ok.